RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -382,67 +387,107 @@ PileupJetIdentifier PileupJetIdAlgo::computeIdVariables(const reco::Jet * jet, f
 			internalId_.ptDCh_     += candPt*candPt;
 			fracCh.push_back(candPtFrac);
 			if( icone < ncones ) { *coneChFracs[icone] += candPt; }
 			sumPtCh                += candPt;
 		}
-		// beta and betastar		
-		if(  icand->trackRef().isNonnull() && icand->trackRef().isAvailable() ) {
-			try { 
-				float tkpt = icand->trackRef()->pt(); 
+		// // beta and betastar		
+		if( icand->charge() != 0 ) {
+		      if (!isPacked){
+			   if(lPF->trackRef().isNonnull() ) { 
+	      			float tkpt = candPt;
 				sumTkPt += tkpt;
 				// 'classic' beta definition based on track-vertex association
-				bool inVtx0 = find( vtx->tracks_begin(), vtx->tracks_end(), reco::TrackBaseRef(icand->trackRef()) ) != vtx->tracks_end();
+				bool inVtx0 = vtx->trackWeight ( lPF->trackRef()) > 0 ;
+					
 				bool inAnyOther = false;
 				// alternative beta definition based on track-vertex distance of closest approach
-				double dZ0 = fabs(icand->trackRef()->dz(vtx->position()));
+				double dZ0 = std::abs(lPF->trackRef()->dz(vtx->position()));
 				double dZ = dZ0;
 				for(reco::VertexCollection::const_iterator  vi=allvtx.begin(); vi!=allvtx.end(); ++vi ) {
-					const reco::Vertex & iv = *vi;
-					if( iv.isFake() || iv.ndof() < 4 ) { continue; }
-					// the primary vertex may have been copied by the user: check identity by position
-					bool isVtx0  = (iv.position() - vtx->position()).r() < 0.02;
-					// 'classic' beta definition: check if the track is associated with any vertex other than the primary one
+				    const reco::Vertex & iv = *vi;
+				    if( iv.isFake() || iv.ndof() < 4 ) { continue; }
+				        // the primary vertex may have been copied by the user: check identity by position
+				        bool isVtx0  = (iv.position() - vtx->position()).r() < 0.02;
+					// 'classic' beta definition: check if the track is associated with
+					// any vertex other than the primary one
 					if( ! isVtx0 && ! inAnyOther ) {
-						inAnyOther = find( iv.tracks_begin(), iv.tracks_end(), reco::TrackBaseRef(icand->trackRef()) ) != iv.tracks_end();
+					  inAnyOther =  vtx->trackWeight ( lPF->trackRef()) <= 0 ;
 					}
 					// alternative beta: find closest vertex to the track
-					dZ = std::min(dZ,fabs(icand->trackRef()->dz(iv.position())));
+					dZ = std::min(dZ,std::abs(lPF->trackRef()->dz(iv.position())));
 				}
 				// classic beta/betaStar
 				if( inVtx0 && ! inAnyOther ) {
-					internalId_.betaClassic_ += tkpt;
+				    internalId_.betaClassic_ += tkpt;
 				} else if( ! inVtx0 && inAnyOther ) {
-					internalId_.betaStarClassic_ += tkpt;
+				    internalId_.betaStarClassic_ += tkpt;
 				}
 				// alternative beta/betaStar
 				if( dZ0 < 0.2 ) {
-					internalId_.beta_ += tkpt;
+				    internalId_.beta_ += tkpt;
 				} else if( dZ < 0.2 ) {
+				    internalId_.betaStar_ += tkpt;
+				}
+			   } else {
+			       if(printWarning-- > 0) { 
+				   edm::LogWarning( "computeIdVariables" ) <<"Invalid reference of track"<<std::endl;
```

Comment:
are these really informative of a problem?
I see the maximum 10 printed just in 20 events (ttbar with PU35).
At this rate, unless there's a bug in this code, this doesn't even deserve to be an info, if the invalid refs are really so frequent.

Please, either change the logic so that this is actually happening much less frequently or remove the printout completely.